lib/tests/test_conflicts.rs

```diff
@@ -2007,6 +2024,214 @@ fn test_update_conflict_from_content_with_long_markers() {
     );
 }
 
+#[test]
+fn test_update_conflict_from_content_no_eol() {
+    let test_repo = TestRepo::init();
+    let store = test_repo.repo.store();
+
+    let path = RepoPath::from_internal_string("file");
+    let base_id = testutils::write_file(store, path, "line 1\nline 2\nline 3\nbase");
+    let left_empty_id =
+        testutils::write_file(store, path, "line 1\nline 2 - left\nline 3\nbase\nleft\n");
+    let right_id = testutils::write_file(store, path, "line 1\nline 2 - right\nline 3\nright");
```

Comment:
nit: better to not use ` - ` as separator? The diff output looks like newline char is missing.